lib/widgets/favorite_toggle_button.dart

```diff
@@ -1,25 +1,38 @@
-import 'package:dears/utils/hooks.dart';
+import 'package:dears/providers/is_wish_listed_provider.dart';
+import 'package:dears/providers/wish_list_ids_provider.dart';
 import 'package:dears/utils/icons.dart';
 import 'package:dears/utils/theme.dart';
 import 'package:flutter/material.dart';
-import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-class FavoriteToggleButton extends HookWidget {
-  final bool initialFavorite;
+class FavoriteToggleButton extends ConsumerStatefulWidget {
+  final int portfolioId;
 
-  const FavoriteToggleButton({
+  const FavoriteToggleButton(
+    this.portfolioId, {
     super.key,
-    required this.initialFavorite,
   });
 
+  @override
+  ConsumerState<FavoriteToggleButton> createState() =>
+      _FavoriteToggleButtonState();
+}
+
+class _FavoriteToggleButtonState extends ConsumerState<FavoriteToggleButton> {
+  Future<void> toggle() async {
+    await ref.read(wishListIdsProvider.notifier).toggle(widget.portfolioId);
+  }
```

Comment:
비동기 처리 과정(add, delete)에서 에러 핸들링도 있으면 좋을 것 같아요 스낵바 같은 거 띄우는 건 어떨까요

Reply:
```suggestion
  Future<void> toggle() async {
    await ref.read(wishListIdsProvider.notifier).toggle(widget.portfolioId);
  }
  
  void showSnackBar() {
    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(
          content: Text("위시리스트를 수정할 수 없습니다"),
        ),
      );
    });
  }

  @override
  Widget build(BuildContext context) {
    final isWishListed =
        ref.watch(isWishListedProvider(widget.portfolioId)).when(
              data: (data) => data,
              error: (error, stackTrace) {
                showSnackBar();
                return false;
              },
              loading: () => false,
            );
    /* ... */
  }
```

이런 방식으로 핸들링하면 좋을 거 같은데, 다른 POST API들도 일괄적으로 에러 핸들링을 해야할 것 같습니다. 추후에 스낵바나 토스트, 다이얼로그 등 하나를 정해서 `Theme`도 적용해서 하면 좋을 것 같습니다. 투두에 남겨놓을게요!!